### ISSUE TITLE
[Waiting for payment] [$250] Migrate index.js to function component

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#16141]
### Filenames
- [ ] [src/components/DisplayNames/index.js](https://github.com/Expensify/App/blob/main/src/components/DisplayNames/index.js)
    - **type:** `React.PureComponent`
    - **has state values:** true
    - **has refs:** true
    - **has context:** false
    - **lifecycle methods:** `componentDidMount`

### Task
- We currently have some class components in our codebase that we would like to refactor to a function component.
- Here's a link with some general advice on how to refactor a class component to a function component: https://react.dev/reference/react/Component#alternatives
- If you need additional guidance, please ask in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q)
- Test for any regressions and verify that there are no breaking changes

**Comment by MelvinBot** [comment]
⚠️ Could not update price automatically because there is no linked Upwork Job ID. The BZ team member will need to update the price manually in Upwork.

**Comment by muxriddinmuqimov77** [comment]
I'd like to work on this issue

**Comment by s-alves10** [comment]
I'd love to work on this.

**Comment by marcaaron** [comment]
Taking this one off HOLD.

**Comment by gedu** [comment]
Hey, I am Edu from Callstack - an expert contributor group.

I can work on this one

**Comment by marcaaron** [comment]
I see @srikarparsi is assigned to this issue. Have they handed it off to you @gedu and should we update the assignment to reflect?

I think it's really up to you and maybe relative to the complexity of this refactor? If it is a small change overall then fixing should be ok. Might want to check to ensure there are no open issues or people working on the bug in parallel. Usually with the obvious bugs people are quite fast to report them.

**Comment by gedu** [comment]
@marcaaron I just saw that was assigned a month ago, didn't notice that was assigned to someone already. 

I wouldn't recommend refactoring this one-to-one from a class component to a functional component. There are some areas that can be improved, and we need to ensure that it doesn't have any negative impact in the future.

On the other hand, the bug fixing this issue is really easy.

**Comment by marcaaron** [comment]
>There are some areas that can be improved, and we need to ensure that it doesn't have any negative impact in the future.

Thanks very much, can you elaborate on the specific concern and which areas could be improved?

>On the other hand, the bug fixing this issue is really easy.

Got it. I think if you have found a bug then let's separate it from the refactor for now and do a small quick PR?

@srikarparsi can you sync up with @gedu and make sure we aren't doubling efforts on this one?

**Comment by gedu** [comment]
@marcaaron Yes.

Firstly, there is an if to handle a Name with and without Tooltip, I would split it into 2 components because the one without tooltip doesn't need any function, refs, or states initialization.
Secondly, the function `getTooltipShiftX` must be in a `useCallback` because deeper in the component `UserDetailsTooltip` it uses a `useMemo` so we need that function stable. In addition, that function is being called using a lambda function `shiftHorizontal={() => getTooltipShiftX(index)}`, so the `useCallback` won't work, we must not use the lambda function and make it stable too.
Lastly, but optional, I would separate into files the different DisplayNames with/without tooltip, just to make the code more readable and avoid having all the logic in the `index.js` file.

Cool, I could post this bug into the `expensify-bugs` channel

**Comment by gedu** [comment]
Hey, @srikarparsi, no problem, I can review the PR

**Comment by strepanier03** [comment]
Looking at this for payment. 

Am I correct in assuming this is $250 payment to both @srikarparsi and @aimane-chnaif?

@marcaaron? Do you have any context into the payment here? Once I know final amounts and who to pay I can make an Upwork job and hire/pay out.

**Comment by strepanier03** [comment]
Hahaha you're right @aimane-chnaif, what a Monday yesterday was 😂.

I'll make the job and hire you to it shortly.

**Comment by strepanier03** [comment]
All right, well I'll try again in a bit to make a job. Upwork is giving me errors no matter how I try to create a job right now. :(

**Comment by strepanier03** [comment]
I've paid this, closed the contract, and submitted the form for the contact list. This is all set now and I'll close it out.
